app/src/main/java/com/akshay/newsapp/ui/activity/MainActivity.kt

```diff
@@ -1,22 +1,22 @@
 package com.akshay.newsapp.ui.activity
 
 import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.akshay.newsapp.R
 import com.akshay.newsapp.adapter.NewsArticlesAdapter
 import com.akshay.newsapp.model.ViewState
-import com.akshay.newsapp.ui.base.BaseActivity
 import com.akshay.newsapp.ui.viewmodel.NewsArticleViewModel
-import com.akshay.newsapp.utils.getViewModel
 import com.akshay.newsapp.utils.observeNotNull
 import com.akshay.newsapp.utils.toast
 import kotlinx.android.synthetic.main.activity_main.*
 import kotlinx.android.synthetic.main.empty_layout.*
 import kotlinx.android.synthetic.main.progress_layout.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class MainActivity : BaseActivity() {
+class MainActivity : AppCompatActivity() {
 
-    private val newsArticleViewModel by lazy { getViewModel<NewsArticleViewModel>() }
+    private val newsArticleViewModel: NewsArticleViewModel by viewModel()
```

Comment:
I'd suggest to keep the `BaseActivity` and have a wrapper / abstraction over `viewModel()` function so our base activities aren't aware of `koin` making it easier to replace it with something else in the future [if the need arises]

Reply:
I am sorry I don't quite get this one. The syntax for injecting a viewmodel() here in case of Koin is such that creating an abstraction over it might be a bit tricky.

Reply:
Umm I see. Let me try a few things on my side

Reply:
Sure. Thanks

Reply:
Here you go:

We just need to modify our existing `ViewModelExtensions.kt` made to work with dagger to use Koin.

```
/**
 * Synthetic sugaring to get instance of [ViewModel] for [AppCompatActivity].
 */
inline fun <reified T : ViewModel> LifecycleOwner.getViewModel(
        qualifier: Qualifier? = null,
        noinline parameters: ParametersDefinition? = null
): Lazy<T> = viewModel(qualifier, parameters)
```

And on the consumer side, it will look like this; that is without any knowledge about Koin 🎉 

` private val newsArticleViewModel: NewsArticleViewModel by getViewModel()`

Reply:
I believe it's the same thing koin is doing behind the scenes. 
I hope this implementation of ours does not create any issue with a major update in Koin's viewModel implementation down the line. 
I have made the requested changes 👍